src/handshakes/abstract-handshake.ts

```diff
@@ -118,9 +119,8 @@ export abstract class AbstractHandshake {
   }
 
   protected getHash(a: bytes, b: bytes): bytes32 {
-    return Buffer.from(
-      hash.sha256().update(Buffer.from([...a, ...b])).digest('hex')
-    );
+    const hashValue = hash.sha256().update(Buffer.from([...a, ...b])).digest();
+    return Buffer.from(hashValue);
```

Comment:
same as above

Reply:
hashValue is number[]. it doesn't have buffer or offset.

Reply:
Could you try it. It's probably Uint8Array but it's displayed as number[] in logs
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array

Reply:
nvm, it's regular Array, wierd. I guess it's fine like this